OrbitGl/App.cpp

```diff
@@ -1152,11 +1152,24 @@ void OrbitApp::InitializeClientTransactions() {
   frame_pointer_validator_client_ =
       std::make_unique<FramePointerValidatorClient>(this,
                                                     transaction_client_.get());
-  process_memory_client_ =
-      std::make_unique<ProcessMemoryClient>(transaction_client_.get());
 }
 
 //-----------------------------------------------------------------------------
 void OrbitApp::FilterFunctions(const std::string& filter) {
   m_LiveFunctionsDataView->OnFilter(filter);
 }
+
+void OrbitApp::GetRemoteMemory(uint32_t pid, uint64_t address, uint64_t size,
+                               const ProcessMemoryCallback& callback) {
+  thread_pool_->Schedule([this, pid, address, size, callback] {
+    auto result = process_manager_->GetProcessMemory(pid, address, size);
+    if (result.has_value()) {
+      main_thread_executor_->Schedule(
+          [memory = std::move(result).value(), callback] { callback(memory); });
```

Comment:
A thought.. since the callback is doing disassembling which could also be done off the main thread, either by moving some of the logic here or calling it without main_thread_executor_ and have it land the update on the main thread (I believe it is Disassemble function in the App.cpp).

Actually looking at it it is already doing it by calling SendToUi so it should be safe to just drop main_thread_executor_.Schedule here.

Reply:
I don't think that this code should change based on the specific callback that is currently passed to it.

Reply:
In which case just move disassemble here?

Reply:
I joined `GetRemoteMemory` and `Disassemble`.